Rotate kubelet client certificates GA, CSR API GA

liggitt opened the pull request:
Docs for:

* https://github.com/kubernetes/enhancements/issues/266
* https://github.com/kubernetes/enhancements/issues/1513

sftim commented:
/milestone 1.19

sftim commented on content/en/docs/tasks/tls/managing-tls-in-a-cluster.md:
```diff
@@ -124,10 +125,13 @@ EOF
 certificate with the "digital signature", "key encipherment", and "server
-auth" key usages. We support all key usages and extended key usages listed
+auth" key usages, signed by the `kubernetes.io/kubelet-serving` signer.
+We support all key usages and extended key usages listed
 [here](https://godoc.org/k8s.io/api/certificates/v1beta1#KeyUsage)
```
Is it correct to revise to:
```suggestion
Kubernetes supports all key usages and extended key usages listed in
[`KeyUsage`](https://godoc.org/k8s.io/api/certificates/v1#KeyUsage)
```
?
Suggesting this mainly to [avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we) but also because I assume that KeyUsage is also going to stabilize into `k8s.io/api/certificates/v1`.

liggitt replied:
I actually dropped that whole sentence in favor of pointing at the signer documentation, which call out the expected usages

sftim commented on content/en/docs/reference/access-authn-authz/certificate-signing-requests.md:
```diff
@@ -315,7 +318,24 @@ Users of the REST API can sign CSRs by submitting an UPDATE request to the `stat
+
+All PEM blocks must have the "CERTIFICATE" label, contain no headers,
+and the encoded data must be a BER-encoded ASN.1 Certificate structure
+as described in section 4 of RFC5280.
```
nit:
```suggestion
as described in section 4 of [RFC5280](https://tools.ietf.org/html/rfc5280).
```

liggitt replied:
linked

sftim commented on content/en/docs/reference/access-authn-authz/certificate-signing-requests.md:
```diff
@@ -315,7 +318,24 @@ Users of the REST API can sign CSRs by submitting an UPDATE request to the `stat
+and the encoded data must be a BER-encoded ASN.1 Certificate structure
+as described in section 4 of RFC5280.
+
+Non-PEM content may appear before or after the CERTIFICATE PEM blocks and is unvalidated,
+to allow for explanatory text as described in section 5.2 of RFC7468.
```
How about adding an example of valid data, prior to base64 encoding.
(the explanatory text could include a note that this whole block is what you base64 encode)?

liggitt replied:
done

savitharaghunathan commented:
/assign

annajung commented:
Hi @liggitt, docs shadow here 👋 just a friendly reminder this PR needs tech and doc review by July 16th.
+ there is a file conflict that needs to be addressed here

sftim commented on content/en/docs/reference/access-authn-authz/certificate-signing-requests.md:
````diff
@@ -315,7 +318,51 @@ Users of the REST API can sign CSRs by submitting an UPDATE request to the `stat
+
+Example certificate content:
+
+```shell
````
nit: this isn't Bourne shell code.

(I can't make a GitHub suggestion for a line that starts with three backticks, otherwise I would)

liggitt replied:
done

sftim commented on content/en/docs/reference/access-authn-authz/certificate-signing-requests.md:
```diff
@@ -315,7 +318,51 @@ Users of the REST API can sign CSRs by submitting an UPDATE request to the `stat
+to allow for explanatory text as described in section 5.2 of RFC7468.
+
+When encoded in JSON or YAML, this field is base-64 encoded.
+A `CertificateSigningRequest` containing the example certificate above would look like this:
```
nit
```suggestion
A CertificateSigningRequest containing the example certificate above would look like this:
```

liggitt replied:
done

liggitt commented:
pushed update addressing comments and fixing rebase issue to https://github.com/liggitt/kubernetes-website/commits/csr-ga but github is not showing it here for some reason

liggitt commented:
/cc @deads2k

deads2k commented:
/lgtm
